### PR TITLE
[gatsby] Compare location object instead of just pathname

### PR DESCRIPTION
**Pull request by paulathevalley**
re: https://github.com/gatsbyjs/gatsby/pull/3802 

This PR compares `window.location` and `location` objects instead of just their pathnames. 

Let me know if there is a better approach. Thanks!

### CODE REVIEW

File: packages/gatsby/cache-dir/production-app.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ import history from "./history"
 window.___history = history
 import emitter from "./emitter"
 window.___emitter = emitter
+import { isEqual } from "lodash"
```
Comment (KyleAMathews) [inline packages/gatsby/cache-dir/production-app.js]: Please use the shallow-compare package as we're already using that so it require any extra code https://github.com/gatsbyjs/gatsby/blob/3ab063feb7997d1c03809cd5ff4d754522df1ac7/packages/gatsby/cache-dir/component-renderer.js#L6

**Comment by KyleAMathews** [comment]
Thanks for jumping on a fix!

**Comment by KyleAMathews** [comment]
Thanks!

**Comment by KyleAMathews** [comment]
Comparing the window and router location objects didn't work as the window object has a bunch of other stuff on it. Instead it seems best to just compare the things we care about directly.

Put up a PR for this https://github.com/gatsbyjs/gatsby/pull/4339

**Comment by jlengstorf** [comment]
Hiya @paulathevalley! 👋

This is definitely late, but on behalf of the entire Gatsby community, I wanted to say thank you for being here.

Gatsby is built by awesome people like you. Let us say “thanks” in two ways:

1.  **We’d like to send you some Gatsby swag.** As a token of our appreciation, you can go to the [Gatsby Swag Store][store] and log in with your GitHub account to get a coupon code good for one free piece of swag. (We’ve got t-shirts and hats, plus some socks that are really razzing our berries right now.)
2.  **If you’re not already part of it, we just invited you to join the Gatsby organization on GitHub.** This will add you to our team of maintainers. You’ll receive an email shortly asking you to confirm. By joining the team, you’ll be able to label issues, review pull requests, and merge approved pull requests.

If you have questions, please don’t hesitate to reach out to us: tweet at [@gatsbyjs][twitter] and we’ll come a-runnin’.

Thanks again! 💪💜

[store]: https://store.gatsbyjs.org
[twitter]: https://twitter.com/gatsbyjs

**Pull request merged**